diff/diffDiff.hpp

```diff
@@ -0,0 +1,126 @@
+/*-----------------------------------------------------------------------------
+
+ Copyright 2017 Hopsan Group
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+
+ The full license is available in the file LICENSE.
+ For details about the 'Hopsan Group' or information about Authors and
+ Contributors see the HOPSANGROUP and AUTHORS files that are located in
+ the Hopsan source code root directory.
+
+-----------------------------------------------------------------------------*/
+
+#ifndef SELTERDIFF_HPP_INCLUDED
+#define SELTERDIFF_HPP_INCLUDED
+
+#include <math.h>
+#include "ComponentEssentials.h"
+#include "ComponentUtilities.h"
+#include <sstream>
+#include <cstring>
+#include <vector>
+#include <string>
+
+using namespace std;
+
+namespace hopsan {
+
+    class diffDiff : public ComponentSignal
+    {
+    private:                         // Private section
+        //Declare local variables
+        double In1;
+        double Out1;
+        double prev;
+        //Declare data pointer variables
+        double *mpIn1, *mpOut1;
+
+        //Declare ports
+
+
+    public:                              //Public section
+        static Component *Creator()
+        {
+            return new diffDiff();
+        }
+
+        //Configure
+        void configure()
+        {
+            //Register constants
+
+
+            //Add ports
+            addInputVariable("In1", "Input", "", 0, &mpIn1);
+            addOutputVariable("Out1", "Output", "", 0, &mpOut1);
```

Comment:
Please just call the ports "in" and "out", to be consistent with other signal components.

Reply:
Looks good to me, just some minor comments. I only commented on diffDiff.hpp, but the same comments apply to both hpp files. If you want to you could also rename the components to "SignalDifference" and "SignalDifferentiator". This would make it easier to move it to the default library later (which I think we should do).

Reply:
Completed in 4d225cd